python/dgl/data/chem/csv_dataset.py

```diff
@@ -39,13 +45,19 @@ class CSVDataset(object):
     Path to store the preprocessed data
     """
 
-    def __init__(self, df, smile2graph=smile2graph, smile_column='smiles',
+    def __init__(self, csvfile_path, smile2graph=smile2graph, smile_column='smiles', id_column=None,
                  cache_file_path="csvdata_dglgraph.pkl"):
         if 'rdkit' not in sys.modules:
             from ...base import dgl_warning
             dgl_warning(
                 "Please install RDKit (Recommended Version is 2018.09.3)")
-        self.df = df
+        if 'pandas' not in sys.modules:
+            from ...base import dgl_warning
+            dgl_warning("Please install pandas")
```

Comment:
Is pandas required or not? If required, you could just throw error in L17

Reply:
Currently in dgl/data/__init__.py having `from .chem import Tox21`, which triggers import rdkit and pandas when importing other dataset. I'll change this.